src/rgw/support/rgw_support_bundle

```diff
@@ -117,32 +133,95 @@ class RGWSupportBundle:
+                    tmp_client_log_dir, size_limit, 'rgw_*')
+                RGWSupportBundle._apply_size_filter(
+                    tmp_client_log_dir, size_limit, 'radosgw*')
+                shutil.rmtree(tmp_svc_files_dir)
+            shutil.rmtree(tmp_client_log_dir)
 
         # copy motr trace log files
         motr_trace_dir = os.path.join(log_dir, 'motr_trace_files')
         if os.path.exists(motr_trace_dir):
+            tmp_motr_trace_dir = os.path.join(RGWSupportBundle._tmp_src, 'motr_trace_files')
             # include the latest 5 log files of motr traces in support bundle
-            list_of_files = filter(lambda f: os.path.isfile(os.path.join(motr_trace_dir, f)),
-                            os.listdir(motr_trace_dir))
-            # sort the files based on last modification time
-            list_of_files = sorted(list_of_files,
-                            key = lambda f: os.path.getmtime(os.path.join(motr_trace_dir, f)),
-                            reverse=True)
-            list_of_files = list_of_files[0:5]
-            for file in list_of_files:
+            latest_files = RGWSupportBundle._get_latest_files(directory=motr_trace_dir)
+            Log.info('Bundle motr trace files')
+            for file in latest_files:
                 infile = os.path.join(motr_trace_dir, file)
-                tmp_motr_trace_dir = os.path.join(RGWSupportBundle._tmp_src, 'motr_trace_files')
-                os.makedirs(tmp_motr_trace_dir, exist_ok=True)
                 outfile = os.path.join(tmp_motr_trace_dir, file)
-                # Convert m0trace file to human readable yaml format
-                SimpleProcess(f'm0trace -i {infile} -Y -o {outfile}.yaml').run()
-                SimpleProcess(f'xz {outfile}.yaml').run() # compress the output file
+                if size_limit:
+                    remaining_size_limit = RGWSupportBundle._get_remaining_folder_size(
+                        RGWSupportBundle._tmp_src, size_limit)
+                    if (os.path.getsize(infile) >= FilterLog._get_size_in_bytes(
+                            remaining_size_limit)):
+                        Log.info('size_limit is full.')
+                        break
+                    os.makedirs(tmp_motr_trace_dir, exist_ok=True)
+                    # Convert m0trace file to human readable yaml format
+                    SimpleProcess(f'm0trace -i {infile} -Y -o {outfile}.yaml').run()
+                    SimpleProcess(f'xz {outfile}.yaml').run()  # compress the output file
+                else:
+                    os.makedirs(tmp_motr_trace_dir, exist_ok=True)
+                    # Convert m0trace file to human readable yaml format
+                    SimpleProcess(f'm0trace -i {infile} -Y -o {outfile}.yaml').run()
+                    SimpleProcess(f'xz {outfile}.yaml').run()  # compress the output file
+
+        # copy addb logs
+        addb_log_path = os.path.join(log_dir, 'addb_files-*')
```

Comment:
Hi @yeshpal-jain-seagate , @nkommuri  
Do we need to collect addb files in support bundle? please confirm.

Reply:
Yes